build-tools/src/main/java/org/elasticsearch/gradle/plugin/StablePluginBuildPlugin.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.plugin;
+
+import org.elasticsearch.gradle.util.GradleUtils;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.SourceSet;
+
+public class StablePluginBuildPlugin implements Plugin<Project> {
+
+    @Override
+    public void apply(Project project) {
+        project.getPluginManager().apply(BasePluginBuildPlugin.class);
+
+        project.getTasks().withType(GeneratePluginPropertiesTask.class).named("pluginProperties").configure(task -> {
+            task.getIsStable().set(true);
+
+            Provider<RegularFile> file = project.getLayout()
+                .getBuildDirectory()
+                .file("generated-descriptor/" + GeneratePluginPropertiesTask.STABLE_PROPERTIES_FILENAME);
+            task.getOutputFile().set(file);
+        });
+
+        final var pluginNamedComponents = project.getTasks().register("pluginNamedComponents", GenerateNamedComponentsTask.class, t -> {
+            SourceSet mainSourceSet = GradleUtils.getJavaSourceSets(project).findByName(SourceSet.MAIN_SOURCE_SET_NAME);
+            t.setPluginClasses(mainSourceSet.getOutput().getClassesDirs());
+            // it has to be compile classpath ?? otherwise libraries which are provided in runtime by es server won't be visible
+            // and we want to scan them too
+            t.setClasspath(project.getConfigurations().getByName(JavaPlugin.COMPILE_CLASSPATH_CONFIGURATION_NAME));
```

Comment:
it has to be compile classpath ?? otherwise libraries which are provided in runtime by es server won't be visible and we want to scan them too

Reply:
Usually a classpath property is the classpath required to run the `GenerateNamedComponentsAction`. I wonder if its needed at all. We probbaly can merge this with pluginClasses can't we?